bindinfo/handle.go

```diff
@@ -492,11 +492,10 @@ func copyBindRecordUpdateMap(oldMap map[string]*bindRecordUpdate) map[string]*bi
 
 func (c cache) getBindRecord(hash, normdOrigSQL, db string) *BindRecord {
 	bindRecords := c[hash]
-	if bindRecords != nil {
```

Comment:
This change makes me feel that the `staticcheck` tool seems kind of silly.
With `if bindRecords != nil` precondition, the compiler is more like to generate better code because it can skip some safety check for array boundary.

Reply:
Well, the purpose of `staticcheck` is clean code, while the purpose of the compiler is performance.
They optimize in different directions and might conflict.

Reply:
Clean code is preferred because the performance doesn't matter too much in most of the cases like here.

Reply:
It's not good if the staticcheck sometimes does give advice by the cost of performance.